receiver/googlecloudmonitoringreceiver/receiver.go

```diff
@@ -5,51 +5,229 @@ package googlecloudmonitoringreceiver // import "github.com/open-telemetry/opent
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+	"time"
 
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.uber.org/zap"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	"google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudmonitoringreceiver/internal"
 )
 
 type monitoringReceiver struct {
-	config *Config
-	logger *zap.Logger
-	cancel context.CancelFunc
+	config         *Config
+	logger         *zap.Logger
+	client         *monitoring.MetricClient
+	metricsBuilder *internal.MetricsBuilder
+	mutex          sync.Mutex
 }
 
 func newGoogleCloudMonitoringReceiver(cfg *Config, logger *zap.Logger) *monitoringReceiver {
 	return &monitoringReceiver{
-		config: cfg,
-		logger: logger,
+		config:         cfg,
+		logger:         logger,
+		metricsBuilder: internal.NewMetricsBuilder(logger),
 	}
 }
 
-func (m *monitoringReceiver) Scrape(ctx context.Context) (pmetric.Metrics, error) {
-	// Dummy use to fix lint errors
-	ctx.Deadline()
-	metrics := pmetric.NewMetrics()
-	m.logger.Debug("Scrape metrics ")
+func (mr *monitoringReceiver) Start(ctx context.Context, _ component.Host) error {
+	var client *monitoring.MetricClient
+	var err error
 
-	return metrics, nil
-}
+	// Lock to ensure thread-safe access to mr.client
+	mr.mutex.Lock()
+	defer mr.mutex.Unlock()
+
+	// If the client is already initialized, return nil
+	if mr.client != nil {
+		return nil
+	}
+
+	// Get service account key file path
+	serAccKey := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
```

Comment:
You should use https://pkg.go.dev/golang.org/x/oauth2/google#FindDefaultCredentials to get credentials, rather than looking up the environment variables yourself.

Reply:
Yes Updated Now